tensorflow_addons/image/cutout_ops.py

```diff
@@ -0,0 +1,166 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Cutout op"""
+
+import tensorflow as tf
+from tensorflow_addons.utils.types import TensorLike, Number
+from tensorflow.python.keras.utils import conv_utils
+
+
+def _get_image_wh(images, data_format):
+    if data_format == "channels_last":
+        image_height, image_width = tf.shape(images)[1], tf.shape(images)[2]
+    else:
+        image_height, image_width = tf.shape(images)[2], tf.shape(images)[3]
+
+    return image_height, image_width
+
+
+def _norm_params(images, mask_size, data_format):
+    mask_size = tf.convert_to_tensor(mask_size)
+    if tf.rank(mask_size) == 0:
+        mask_size = tf.stack([mask_size, mask_size])
+    data_format = conv_utils.normalize_data_format(data_format)
+    image_height, image_width = _get_image_wh(images, data_format)
+    return mask_size, data_format, image_height, image_width
+
+
+def random_cutout(
+    images: TensorLike,
+    mask_size: TensorLike,
+    constant_values: Number = 0,
+    seed: Number = None,
+    data_format: str = "channels_last",
+) -> tf.Tensor:
+    """Apply cutout (https://arxiv.org/abs/1708.04552) to images.
+
+    This operation applies a (2 * mask_height x 2 * mask_width) mask of zeros to
+    a random location within `img`. The pixel values filled in will be of the
+    value `replace`. The located where the mask will be applied is randomly
+    chosen uniformly over the whole images.
+
+    Args:
+      images: A tensor of shape
+        (batch_size, height, width, channels)
+        (NHWC), (batch_size, channels, height, width)(NCHW).
+      mask_size: Specifies how big the zero mask that will be generated is that
+        is applied to the images. The mask will be of size
+        (2 * mask_height x 2 * mask_width).
```

Comment:
`mask_size` should be the shape of the mask. Here the shape of the true mask is `(2 * mask_height x 2 * mask_width)` which might confuse users. Could you change that to `(mask_height x mask_width)`? You can throw an error if the width and height cannot be divided by 2.